apps/cassandra.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+)
+
+type MetricsReceiverCassandra struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,url"`
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+
+	CollectJVMMetics *bool `yaml:"collect_jvm_metrics"`
```

Comment:
I'm not a big fan of collecting metrics that collide with the JVM receiver by default when users aren't expecting them, at least until we resolve the separate issues around how to handle monitoring for multiple processes.

I'd like to suggest a third alternative to consider: Collect the JVM metrics, but consider prefixing them with `cassandra.` I think that's a better UX now, but it may conflict with what we want to do around extra resource attributes in the future.

Reply:
The decision to have cassandra report JVM metrics is inline with how the OTel JMX receiver works and user expectations around cassandra monitoring (where JVM GC is a big portion of it).

When you mix user expectations (monitoring Cassandra needs JVM GC monitoring), concerns around having multiple JVM GC metrics with different names, etc.  I think having cassandra default to collecting JVM metrics is the right thing.   The only concern I have is whether we add a label to them denoting "this was cassandra".  That shouldn't be necessary given we'll (eventually) have a label on all JVM metrics denoting which resource they came from.

TL;DR:   I'm not a fan of splitting JVM from Cassandra.  This matches user expectation and has precedence in other solutions.

Reply:
I like that idea. Also commented about it in https://github.com/GoogleCloudPlatform/ops-agent/pull/220#discussion_r724711297. Sam seems to be onboard too.